File: Codigos Java/Archivos/pruebaArchivos.java
package Archivos;
import java.io.*;
import java.util.ArrayList;

public class pruebaArchivos {
    private ArrayList<String> contenidoArchivo = new ArrayList<>();
    private ArrayList<Obra> listaEntrada = new ArrayList<>();
    private final String rutaWindows = "C:\\Users\\Erick\\Documents\\Programacion\\Nueva carpeta\\Jorge\\src\\Archivos\\datos.txt";

    public ArrayList<String> getContenidoArchivo() {
        return contenidoArchivo;
    }

    public ArrayList<Obra> getListaEntrada() {
        return listaEntrada;
    }

    public void leerArchivo() {
        LeerArchivo leer = new LeerArchivo();
        this.contenidoArchivo =
                leer.leerArchivo(this.rutaWindows);
        for (int i = 0; i < this.contenidoArchivo.size(); i++) {
            String linea = this.contenidoArchivo.get(i);
            String elementosLinea[] = linea.split(";");
            try {
                float obrero1 = Float.parseFloat(elementosLinea[0]);
                float obrero2 = Float.parseFloat(elementosLinea[1]);
                float obrero3 = Float.parseFloat(elementosLinea[2]);
                float obrero4 = Float.parseFloat(elementosLinea[3]);
                listaEntrada.add(new Obra(obrero1, obrero2, obrero3, obrero4));
            } catch (NumberFormatException ex) {
                ex.printStackTrace();
                listaEntrada.add(new Obra(0, 0, 0, 0));
            }
        }
    }
    public void escribirArchivo(ArrayList<Obra> lista){
        String archivo = "resultados.csv";
        File f = new File(archivo);
        try{
            FileWriter w = new FileWriter(f);
            BufferedWriter bw = new BufferedWriter(w);
            PrintWriter wr = new PrintWriter(bw);
            wr.write("");
            wr.append("obrero1, obrero2, obrero3, obrero4, tiempoDemoradoJuntos\n");
            for (int i=0; i < lista.size(); i++){
                wr.append(lista.get(i).getObrero1() + "," + lista.get(i).getObrero2() +"," + lista.get(i).getObrero3() +
                        "," + lista.get(i).getObrero4() + "," + lista.get(i).getTiempoDemoradoJuntos() + "\n");
            }
            wr.close();
            bw.close();
        }catch(IOException e){

        }
    }

    public static void main (String args[]){
        pruebaArchivos prueba = new pruebaArchivos();
        prueba.leerArchivo();
        for (int i=0; i < prueba.getListaEntrada().size(); i++){
            System.out.println(prueba.getListaEntrada().get(i).getObrero1() +
                    "\t" + prueba.getListaEntrada().get(i).getObrero2() +
                    "\t" + prueba.getListaEntrada().get(i).getObrero3() +
                    "\t" + prueba.getListaEntrada().get(i).getObrero4() +
                    "\t" + prueba.getListaEntrada().get(i).getTiempoDemoradoJuntos());
        }
        prueba.escribirArchivo(prueba.getListaEntrada());
    }
}

File: Codigos Python/Archivos/pruebaArchivos.py
from abc import ABCMeta


class Obra(ABCMeta):

    # Metodo de clase
    @staticmethod
    def calcularTiempoDemoradoJuntos(obrero1, obrero2, obrero3, obrero4):
        tiempoDemorado = 1 / ((1 / obrero1) + (1 / obrero2) + (1 / obrero3) + (1 / obrero4))
        tiempoDemorado = round(tiempoDemorado, 2)
        return tiempoDemorado


class pruebaArchivos:
    def leerArchivo(self, archivo):
        file = open(archivo, 'r')
        lineas = []
        lineas_archivo = []
        for linea in file.readlines():
            lineas.append(linea.replace('\n', '').split(";"))
        file.close()
        for f in range(0, len(lineas)):
            try:
                lineas_archivo.append([int(lineas[f][0]), int(lineas[f][1]), int(lineas[f][2]), int(lineas[f][3])])
            except ValueError:
                lineas_archivo.append([0, 0, 0, 0])
        return lineas_archivo

    def Obreros(self, lista):
        resultados = []
        for f in range(0, len(lista)):
            resultados.append(Obra.calcularTiempoDemoradoJuntos(lista[f][0], lista[f][1], lista[f][2], lista[f][3]))
        return resultados

    def guardarResultados(self, entrada, resultados):
        file = open("resultados.txt", 'w')
        file.write('Obrero1;Obrero2;Obrero3,Obrero4, tiempoQueTardaranEnRelaizarLaOBra\n')
        for f in range(0, len(entrada)):
            linea = str(entrada[f][0]) + ';' + str(entrada[f][1]) + ';' + str(entrada[f][2]) + ';' + str(
                entrada[f][3]) + ';' + str(resultados[f]) + '\n'
            file.write(linea)
        file.close()


if __name__ == "__main__":
    prueba = pruebaArchivos()
    archivo = []
    archivo = prueba.leerArchivo("datos.txt")
    resultado = prueba.Obreros(archivo)
    prueba.guardarResultados(archivo, resultado)
